Source/Core/Core/AchievementManager.cpp

```diff
@@ -55,614 +60,155 @@ void AchievementManager::SetUpdateCallback(UpdateCallback callback)
-    {
-      m_image_queue.EmplaceItem([this, index] {
-        std::string current_name, name_to_fetch;
-        {
-          std::lock_guard lock{m_lock};
-          if (m_game_data.num_achievements <= index)
-          {
-            INFO_LOG_FMT(
-                ACHIEVEMENTS,
-                "Attempted to fetch unlocked badge for index {} after achievement list cleared.",
-                index);
-            return;
-          }
-          const auto& achievement = m_game_data.achievements[index];
-          const auto unlock_itr = m_unlock_map.find(achievement.id);
-          if (unlock_itr == m_unlock_map.end())
-          {
-            ERROR_LOG_FMT(
-                ACHIEVEMENTS,
-                "Attempted to fetch unlocked badge for achievement id {} not in unlock map.",
-                index);
-            return;
-          }
-          name_to_fetch = achievement.badge_name;
-          current_name = unlock_itr->second.unlocked_badge.name;
-        }
-        if (current_name == name_to_fetch)
-          return;
-        rc_api_fetch_image_request_t icon_request = {.image_name = name_to_fetch.c_str(),
-                                                     .image_type = RC_IMAGE_TYPE_ACHIEVEMENT};
-        Badge fetched_badge;
-        if (RequestImage(icon_request, &fetched_badge) == ResponseType::SUCCESS)
-        {
-          INFO_LOG_FMT(ACHIEVEMENTS, "Successfully downloaded unlocked achievement badge id {}.",
-                       name_to_fetch);
-          std::lock_guard lock{m_lock};
-          if (m_game_data.num_achievements <= index)
-          {
-            INFO_LOG_FMT(ACHIEVEMENTS,
-                         "Fetched unlocked badge for index {} after achievement list cleared.",
-                         index);
-            return;
-          }
-          const auto& achievement = m_game_data.achievements[index];
-          const auto unlock_itr = m_unlock_map.find(achievement.id);
-          if (unlock_itr == m_unlock_map.end())
-          {
-            ERROR_LOG_FMT(ACHIEVEMENTS,
-                          "Fetched unlocked badge for achievement id {} not in unlock map.", index);
-            return;
-          }
-          if (name_to_fetch != achievement.badge_name)
-          {
-            INFO_LOG_FMT(
-                ACHIEVEMENTS,
-                "Requested outdated unlocked achievement badge id {} for achievement id {}.",
-                name_to_fetch, current_name);
-            return;
-          }
-          unlock_itr->second.unlocked_badge.badge = std::move(fetched_badge);
-          unlock_itr->second.unlocked_badge.name = std::move(name_to_fetch);
-        }
-        else
-        {
-          WARN_LOG_FMT(ACHIEVEMENTS, "Failed to download unlocked achievement badge id {}.",
-                       name_to_fetch);
-        }
-
-        m_update_callback();
-      });
-    }
-    if (unlock_status.locked_badge.name != badge_name_to_fetch)
+    auto bucket = achievement_list->buckets[bx];
```

Comment:
Not sure what is in `buckets` but it isn't a pod type, so you should store by reference to avoid a copy:

`auto& bucket = ...`